lib/slack_interactor.ex

```diff
@@ -0,0 +1,44 @@
+defmodule SlackInteractor do
+  @moduledoc "Helper module for posting messages to slack"
+  require Logger
+
+  @doc "Accepts a JSON object from a github webhook POST"
+  def notify_from_github_event(%{"action" => action, "pull_request" => pull_request}) do
+    notify_pull_request(action, pull_request)
+  end
+  def notify_from_github_event(%{"zen" => zen}) do
+    Logger.info(zen)
+    :ok
+  end
+  def notify_from_github_event(_) do
+    Logger.debug("Unhandled event")
+    :ok
+  end
+
+  defp notify_pull_request("opened", pull_request) do
+    %{"html_url" => url, "title" => title, "user" => %{"login" => username}} = pull_request
```

Comment:
Yes, if we match on "opened" action I would prefer to raise a pattern match failure on line 19 than silently fall through to the default case (which would happen if we matched in the function head).